addons/knowledge/static/src/js/knowledge_controller.js

```diff
@@ -218,33 +219,77 @@ const KnowledgeFormController = FormController.extend({
     /**
      * @param {Object} data
      * @param {integer} data.article_id
-     * @param {String} data.category
+     * @param {String} data.oldCategory
+     * @param {String} data.newCategory
      * @param {integer} [data.target_parent_id]
      * @param {integer} [data.before_article_id]
      * @param {Function} data.onSuccess
      * @param {Function} data.onReject
      */
-    _move: async function (data) {
-        const params = {
-            private: data.category === 'private'
+    _confirmMove: async function (data) {
+        data['params'] = {
+            private: data.newCategory === 'private'
         };
         if (typeof data.target_parent_id !== 'undefined') {
             params.parent_id = data.target_parent_id;
         }
         if (typeof data.before_article_id !== 'undefined') {
             params.before_article_id = data.before_article_id;
         }
-        const result = await this._rpc({
+        if (data.newCategory == data.oldCategory || data.newCategory == 'shared') {
```

Comment:
Actually, we cannot move articles into shared, unless the oldCategory was already "shared".
I think we should ensure this here but also in move_to method on article model. (we have to make the check here to avoid useless rpc to the server. But we should also avoid this behaviour in backend anyway.)

Reply:
done